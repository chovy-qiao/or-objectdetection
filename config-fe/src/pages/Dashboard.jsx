import React from "react";
import { connect } from "react-redux";
import {
  LoadVideoSources,
  AddVideoSource
} from "store/modules/video_sources/sourcesSlice";

import { Grid, Typography } from "@material-ui/core";
import Popup from "features/custom/Popup";
import CustomVideoCard from "features/card/customVideoCard";
import { withStyles } from "@material-ui/core/styles";

const mapStateToProps = (state) => ({
  feeds: state.sources.videoSources
});
const mapDispatch = { LoadVideoSources, AddVideoSource };

const useStyles = (theme) => ({
  title: {
    marginBottom: 22
  }
});

class Dashboard extends React.Component {
  componentDidMount() {
    this.props.LoadVideoSources();
  }

  render() {
    const { classes, feeds } = this.props;
    return (
      <div>
        <Typography className={classes.title} variant="h4" color="primary">
          VIDEO SOURCES
        </Typography>

        <Grid container spacing={1}>
          <Grid item xs={12} sm={6} md={4} lg={3} xl={2} heigth="100%">
            <Popup />
          </Grid>

          {feeds.map((value, index) => {
            return (
              <Grid item xs={12} sm={6} md={4} lg={3} xl={2} key={index}>
                <CustomVideoCard
                  Id={value.id}
                  SourceName={value.name}
                  SubName={value.location}
                  Descripton={value.description}
                  SourceType={value.feed_type}
                />
              </Grid>
            );
          })}
        </Grid>
      </div>
    );
  }
}

export const DashboardWithStyles = withStyles(useStyles)(Dashboard);

export default connect(mapStateToProps, mapDispatch)(DashboardWithStyles);
