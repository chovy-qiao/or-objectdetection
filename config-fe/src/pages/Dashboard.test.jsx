import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { DashboardWithStyles } from "./Dashboard";
import { MemoryRouter } from "react-router-dom";

const renderDashboard = ({ LoadVideoSources = jest.fn(), feeds = [] } = {}) => {
  return render(
    <MemoryRouter>
      <DashboardWithStyles feeds={feeds} LoadVideoSources={LoadVideoSources} />
    </MemoryRouter>
  );
};
describe("Dashboard component", () => {
  it("matches the snapshot", () => {
    const { container } = renderDashboard();
    expect(container).toMatchSnapshot();
  });
});
