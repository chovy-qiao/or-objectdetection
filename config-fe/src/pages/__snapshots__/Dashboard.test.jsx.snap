// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component matches the snapshot 1`] = `
<div>
  <div>
    <h4
      class="MuiTypography-root Dashboard-title-1 MuiTypography-h4 MuiTypography-colorPrimary"
    >
      VIDEO SOURCES
    </h4>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2"
        heigth="100%"
      >
        <div>
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-16 MuiPaper-outlined MuiPaper-rounded"
          >
            <div
              class="MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-avatar"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-14 MuiAvatar-colorDefault"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiCardHeader-title makeStyles-title-15 MuiTypography-h6 MuiTypography-displayBlock"
                >
                  New Source
                </span>
                <span
                  class="MuiTypography-root MuiCardHeader-subheader makeStyles-title-15 MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                >
                  Click here to add a new source
                </span>
              </div>
            </div>
            <div
              class="makeStyles-mediaContainer-17"
            >
              <div
                class="MuiCardMedia-root makeStyles-media-13"
                style="background-image: url(plusicon.png);"
                title="plusIcon"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
